lib/SIL/IR/SILFunctionType.cpp

```diff
@@ -2385,7 +2385,7 @@ struct DeallocatorConventions : Conventions {
 
   ParameterConvention
   getIndirectSelfParameter(const AbstractionPattern &type) const override {
-    llvm_unreachable("Deallocators do not have indirect self parameters");
+    return ParameterConvention::Indirect_In;
```

Comment:
I'm curious about why this change was needed.

Reply:
This is for supporting deinits which get the self argument as indirect argument, which is the case for move-only types.